spec/support/features/irs_attempts_api_tracking_helper.rb

```diff
@@ -29,7 +29,11 @@ def irs_attempts_api_tracked_events(timestamp:)
   def stub_attempts_tracker
     irs_attempts_api_tracker = FakeAttemptsTracker.new
 
-    allow(controller).to receive(:irs_attempts_api_tracker).and_return(irs_attempts_api_tracker)
+    if defined?(controller)
+      allow(controller).to receive(:irs_attempts_api_tracker).and_return(irs_attempts_api_tracker)
```

Comment:
I'd use `respond_to?` here? but otherwise same same, I'm not sure any strong reason to use one or the other here
```suggestion
    if respond_to?(:controller)
      allow(controller).to receive(:irs_attempts_api_tracker).and_return(irs_attempts_api_tracker)
```